test/libyul/yulOptimizerTests/fullSuite/ssaReverseComplex.yul

```diff
@@ -0,0 +1,23 @@
+{
+    let a := mload(0)
+    let b := mload(1)
+    if mload(2) {
+        a := mload(b)
+        b := mload(a)
+        a := mload(b)
+        b := mload(a)
+    }
+    mstore(a, b)
+}
+// ----
+// fullSuite
+// {
+//     let a := mload(0)
+//     let b := mload(1)
+//     if mload(2)
+//     {
+//         a := mload(mload(mload(b)))
+//         b := mload(a)
```

Comment:
Ah, right, it can't inline ``a`` as well, since it needs it below - still: works nicely!